api/v1alpha1/mustgather_types.go

```diff
@@ -55,12 +47,38 @@ type MustGatherSpec struct {
 	// +kubebuilder:validation:Format=duration
 	MustGatherTimeout metav1.Duration `json:"mustGatherTimeout,omitempty"`
 
+	// The target location for the must-gather bundle to be uploaded to.
+	// If not specified, the bundle will not be uploaded.
+	// +kubebuilder:validation:Optional
+	UploadTarget *UploadTargetSpec `json:"uploadTarget,omitempty"`
+}
+
+// SFTPSpec defines the desired state of SFTPSpec
+type SFTPSpec struct {
+	// The is of the case this must gather will be uploaded to
+	// +kubebuilder:validation:Required
+	CaseID string `json:"caseID"`
+
+	// the secret container a username and password field to be used to authenticate with red hat case management systems
+	// +kubebuilder:validation:Required
+	CaseManagementAccountSecretRef corev1.LocalObjectReference `json:"caseManagementAccountSecretRef"`
+
 	// A flag to specify if the upload user provided in the caseManagementAccountSecret is a RH internal user.
 	// See documentation for further information.
 	// +kubebuilder:default:=true
 	InternalUser bool `json:"internalUser,omitempty"`
 }
 
+// UploadTargetSpec defines the desired state of UploadTargetSpec
+type UploadTargetSpec struct {
+	// +kubebuilder:validation:Enum=SFTP
+	// +kubebuilder:validation:Required
+	Type string `json:"type"`
```

Comment:
this needs a unionDiscriminator and `// +required` too.

Reply:
Please add a new `type UploadType` and we also need to add `const UploadTypeSFTP`, a good example to refer would be https://github.com/openshift/api/blob/5d856d3e835453391cc5491910e089d2cfbb4752/config/v1/types_node.go#L77-L85.